### PR TITLE
tidb-server: return 1 while failing to set cpu affinity

### LINKED ISSUE #30196: invalid `-affinity-cpus` returns with wrong retcode
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

`./bin/tidb-server -affinity-cpus x; echo $?`

### 2. What did you expect to see? (Required)

1

### 3. What did you see instead (Required)

0

### PR DESCRIPTION
**Pull request by zhangyangyu**
### What problem does this PR solve?

Issue Number: close #30196 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by bb7133 — APPROVED** [review]
LGTM

File: tidb-server/main.go [hunk 1]
```diff
@@ -259,15 +254,15 @@ func setCPUAffinity() {
 			c, err := strconv.Atoi(af)
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "wrong affinity cpu config: %s", *affinityCPU)
-				exit()
+				os.Exit(1)
```
Comment (mjonss) [inline tidb-server/main.go:257]: This change removes the `syncLog()` call, is that intended?

**Review by mjonss — APPROVED** [review]
LGTM

But I wonder if the `syncLog()` removal was intended?

File: tidb-server/main.go (hunk 1 above)
Comment (zhangyangyu) [reply tidb-server/main.go:257]: yes. I think no need to `syncLog` here. If it's necessary, many other places need too.

Comment (lysu) [reply tidb-server/main.go:257]: +1, sync on stdout/stderr seems be useless https://github.com/uber-go/zap/issues/370#issuecomment-307152090, maybe we can delete them(both in linux and newest macos)

**Pull request merged**